lib/Runtime/Language/ModuleNamespace.h

```diff
@@ -6,14 +6,78 @@
 
 namespace Js
 {
-    /// Module namespace object
-    // This object, while the object is special that the properties are indirect bindings referencing variable
-    // definitions from different module, we can add the properties to the object and [[get]] them accordingly.
     class ModuleNamespace : public DynamicObject
     {
+        typedef JsUtil::BaseDictionary<PropertyId, ModuleNameRecord, RecyclerLeafAllocator, PowerOf2SizePolicy> UnambiguousExportMap;
+
+    protected:
+        DEFINE_VTABLE_CTOR(ModuleNamespace, DynamicObject);
+        DEFINE_MARSHAL_OBJECT_TO_SCRIPT_CONTEXT(ModuleNamespace);
+
+    protected:
+        ModuleNamespace(ModuleRecordBase* moduleRecord, DynamicType * type);
+        static ModuleNamespace* New(ModuleRecordBase* moduleRecord);
     public:
-        ModuleNamespace();
-    private:
+        static ModuleNamespace* GetModuleNamespace(ModuleRecordBase* moduleRecord);
+
+        static bool Is(Var ns) { return ModuleNamespace::Is(ns); }
 
+        //virtual int GetPropertyCount() override;
```

Comment:
These are unimplemented feature. We'll uncomment these in upcoming changes.

---
In reply to: [55074978](https://github.com/Microsoft/ChakraCore/pull/432#discussion_r55074978) [](ancestors = 55074978)